src/prefect/environments/storage/docker.py

```diff
@@ -533,59 +550,40 @@ def create_dockerfile_object(self, directory: str) -> str:
             health_file.write(healthcheck)
 
-        if self.dockerfile:
-            with open(self.dockerfile, "r") as contents:
-                base_commands = textwrap.indent("\n" + contents.read(), prefix=" " * 16)
-        else:
-            base_commands = "FROM {base_image}".format(base_image=self.base_image)
+        # Escape the healthcheck location
+        healthcheck_loc = (
+            healthcheck_loc.replace("\\", "/") if self.dockerfile else "healthcheck.py"
+        )
+
+        # Generate the command to run the healthcheck
+        healthcheck_run = ""
+        if not self.ignore_healthchecks:
+            flow_file_paths = ", ".join(['"{}"'.format(k) for k in self.flows.values()])
+            python_version = (sys.version_info.major, sys.version_info.minor)
+            healthcheck_run = (
+                f"RUN python {self.prefect_directory}/healthcheck.py "
+                f"'[{flow_file_paths}]' '{python_version}'"
+            )
 
         file_contents = textwrap.dedent(
-            """\
-            {base_commands}
+            f"""
+            {multiline_indent(base_commands, 12)}
 
             RUN pip install pip --upgrade
-            {installation_commands}
+            {multiline_indent(installation_commands, 12)}
             {pip_installs}
 
-            RUN mkdir -p {prefect_dir}/
-            {copy_flows}
-            COPY {healthcheck_loc} {prefect_dir}/healthcheck.py
-            {copy_files}
-
-            {env_vars}
-            {final_commands}
-            """.format(
-                base_commands=base_commands,
-                installation_commands=installation_commands,
-                pip_installs=pip_installs,
-                copy_flows=copy_flows,
-                healthcheck_loc=healthcheck_loc.replace("\\", "/")
-                if self.dockerfile
-                else "healthcheck.py",
-                copy_files=copy_files,
-                env_vars=env_vars,
-                prefect_dir=self.prefect_directory,
-                final_commands=final_commands,
-            )
-        )
+            RUN mkdir -p {self.prefect_directory}/
+            {multiline_indent(copy_flows, 12)}
+            COPY {healthcheck_loc} {self.prefect_directory}/healthcheck.py
+            {multiline_indent(copy_files, 12)}
 
-        # append the line that runs the healthchecks
-        # skip over for now if storing flow as file
-        if not self.ignore_healthchecks:
-            file_contents += textwrap.dedent(
-                """
-
-                RUN python {prefect_dir}/healthcheck.py '[{flow_file_paths}]' '{python_version}'
-                """.format(
-                    flow_file_paths=", ".join(
-                        ['"{}"'.format(k) for k in self.flows.values()]
-                    ),
-                    python_version=(sys.version_info.major, sys.version_info.minor),
-                    prefect_dir=self.prefect_directory,
-                )
-            )
```

Comment:
This string is now always templated but is empty if we're ignoring healthchecks which reduces the nesting and makes it more readable